tensorflow/core/kernels/mkl_conv_ops.cc

```diff
@@ -42,26 +43,239 @@ limitations under the License.
 #include "tensorflow/core/util/mkl_util.h"
 
 #ifndef INTEL_MKL_ML
-
 #include "mkldnn.hpp"
 
 using mkldnn::prop_kind;
 using mkldnn::stream;
-
-using mkldnn::convolution_direct;
 using mkldnn::convolution_forward;
+using mkldnn::convolution_direct;
+
 #else
 #include "mkl_dnn.h"
 #include "mkl_dnn_types.h"
 #endif
 
 namespace tensorflow {
 
+#ifndef INTEL_MKL_ML
+
+struct ConvFwdDimensions {
+  memory::dims src_dims;
+  memory::dims filter_dims;
+  memory::dims bias_dims;
+  memory::dims dst_dims;
+  memory::dims strides;
+  memory::dims dilations;
+  memory::dims padding_l;
+  memory::dims padding_r;
+
+  ConvFwdDimensions(memory::dims src_d,
+      memory::dims filter_d, memory::dims bias_d,
+      memory::dims dst_d, memory::dims strides,
+      memory::dims dilations, memory::dims pl,
+      memory::dims pr) : src_dims(src_d), filter_dims(filter_d),
+          bias_dims(bias_d), dst_dims(dst_d), strides(strides),
+          dilations(dilations), padding_l(pl), padding_r(pr) {
+  }
+};
+
+template <typename T>
+class Conv2DFwd : public DnnOp {
+ public:
+  explicit Conv2DFwd(const ConvFwdDimensions& convFwdDims) {
+    fwd_stream_.reset(new stream(stream::kind::eager));
+    // create conv primitive
+    if (conv_fwd_ == nullptr) {
+      Setup(convFwdDims);
+    }
+  }
+
+  ~Conv2DFwd() {}
+
+  void Execute() {}
```

Comment:
Is this overload needed?

Reply:
No, it is not needed and thus has been removed.